browser/tor/tor_launcher_factory.cc

```diff
@@ -127,28 +128,172 @@ void TorLauncherFactory::RemoveObserver(tor::TorProfileServiceImpl* service) {
 }
 
 void TorLauncherFactory::OnTorLauncherCrashed() {
-  LOG(ERROR) << "Tor Launcher Crashed";
+  LOG(INFO) << "Tor Launcher Crashed";
   is_starting_ = false;
   for (auto& observer : observers_)
     observer.NotifyTorLauncherCrashed();
 }
 
 void TorLauncherFactory::OnTorCrashed(int64_t pid) {
-  LOG(ERROR) << "Tor Process(" << pid << ") Crashed";
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  LOG(INFO) << "Tor Process(" << pid << ") Crashed";
   is_starting_ = false;
+  is_connected_ = false;
   for (auto& observer : observers_)
     observer.NotifyTorCrashed(pid);
+  KillTorProcess();
+  // Post delayed relaucn for control to stop
+  content::GetUIThreadTaskRunner({})
+    ->PostDelayedTask(FROM_HERE,
```

Comment:
nit - 4 spaces for continuation of a line

Reply:
fixed in https://github.com/brave/brave-core/pull/4802/commits/60f91be1399cfdf0df955bca2bd143487b20a188